pkg/network/http/monitor.go

```diff
@@ -171,6 +188,16 @@ func (m *Monitor) GetHTTPStats() map[Key]RequestStats {
 	return <-reply
 }
 
+func (m *Monitor) GetStats() (map[string]int64, error) {
+	if len(m.lastTelemetry) == 0 {
+		return map[string]int64{}, fmt.Errorf("error: could not get telemetry data")
+	}
+	telem := <-m.lastTelemetry
```

Comment:
should `lastTelemetry` be a `*telemetry` object so that `GetStats` can be queried by multiple clients & still return the most recent telemetry?

Reply:
Well, the issue is that, as we're resetting the state of the "state keeper", we'll be in conflict in terms of what values we're seeing. We're only calling `GetStats` after calling `GetHTTPStats`, so I think we'll be replacing the pointer value with the most recent one anyways.

However, I am not against changing the channel to something else, I just think we'll end up with the same data.

Reply:
You could just use functions from `atomic` package to load the values from the last delta stored as a `telemetry` object in `GetStats`. I think this would be simpler.

Reply:
I'll switch to this, although, for the record, I think it still doesn't fix the problem with this "in two times" fetching of the stats. As for now, our usage of the http submodule is pretty straightforward (I think we don't even need to use atomics as we are expecting to be used a particular way from the tracer anyways).

I kind of feel I was trying to fix a concern with the channel (like, we kind of ensure we're being used the way we expect it to), and using pointers and atomics feels like if we're trying to do defensive programming, **but we still expect to be used the way we're intended to be used**.

IMO, with such expectation, we better be using pointers without atomics at all.

Reply:
I agree that atomics are a clearer way to implement this. The channel stuff feels a bit too clever, and it is hard to understand the purpose at first glance.

Without solving the telemetry/stats synchronization problem, I think having this function return the most recent calculated values is OK.